docs/ha-considerations.md

```diff
@@ -83,23 +83,32 @@ There are some placeholders in `bash` variable style to fill in:
 
 The above `keepalived` configuration uses a health check script `/etc/keepalived/check_apiserver.sh` responsible for making sure that on the node holding the virtual IP the API Server is available. This script could look like this:
 
-```
+```bash
 #!/bin/sh
 
 errorExit() {
     echo "*** $*" 1>&2
     exit 1
 }
 
-curl --silent --max-time 2 --insecure https://localhost:${APISERVER_DEST_PORT}/ -o /dev/null || errorExit "Error GET https://localhost:${APISERVER_DEST_PORT}/"
-if ip addr | grep -q ${APISERVER_VIP}; then
-    curl --silent --max-time 2 --insecure https://${APISERVER_VIP}:${APISERVER_DEST_PORT}/ -o /dev/null || errorExit "Error GET https://${APISERVER_VIP}:${APISERVER_DEST_PORT}/"
+# replace this with the virtual IP address negotiated between the `keepalived` cluster hosts
+APISERVER_VIP=${APISERVER_VIP_VALUE}
+# replace this with the port through which Kubernetes will talk to the API Server
+APISERVER_DEST_PORT=${APISERVER_DEST_PORT_VALUE}
+
+url="https://localhost:$APISERVER_DEST_PORT"
+if ip addr | grep -q $APISERVER_VIP; then
+        url="https://$APISERVER_VIP:$APISERVER_DEST_PORT"
 fi
+http_code=$(curl --output /dev/null --silent --write-out "%{http_code}" --insecure --max-time 2 ${url})
+err_code=$?
```

Comment:
uppercase ERR_CODE

Reply:
The err_code variable is unnecessary. It's really cosmetical, but I'd probably prefer to see the code structured like this:
```
HTTP_CODE=$(curl --output /dev/null --silent --write-out "%{http_code}" --insecure --max-time 2 ${URL}) || errorExit "Error get '$URL'."

# some words explaining why code 403 is considered OK
test "$HTTP_CODE" = "200" -o "$HTTP_CODE" = "403" || errorExit "Error get '$URL', status code '$HTTP_CODE'."
```

There's no need to track a nonzero exit code, hence no need to keep it in a variable and add an additional line for checking it.

There's no need for `[ ... ] || [ ... ]` constructions, an OR expression is something a single `test` command is perfectly capable of (I personally prefer to write `test` rather than `[ ... ]`, but that's a matter of taste.

Whenever quoting status codes and the like in shell scripts using single quotes is recommendable for better readability of the output and making spelling errors (resulting in empty variable values) easier to find.